networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -225,6 +225,14 @@ where
 {
     while let Some(commitment) = receiver.recv().await {
         println!("Got commitment: {:?}", commitment);
+        let max_height = settlement_client.get_max_tolerable_block_height().await?;
```

Comment:
This is probably naive and wasteful to do once per block.

If the `post_block_commitment` call below can be expected to provide back-pressure, we can implement simple batching by draining the receiver to form the next batch.

Reply:
1. The intent I had is to have the client crate simply call and stream events from the contract--keeping it fairly raw.
2. The layer or an intermediate layer to the one you're working on is where I would envision backpressure to be handled. 
3. You would not use multiple `post_block_commitment`calls, you would use `post_batch_block_commitment` (which sends in one transaction). 
4. I would do this when you detect a condition for draining such as all sent block heights have been either accepted or rejected. I would then  call `get_max_tolerable_block_height().await?` out to that height.

Reply:
Also, when we start addressing reorgs we might want to expose the contract `getCommittedBlockAtHeight` through this client so that we avoid trying to commit over where we had already made mistakes.